### PR TITLE
Arizona Dependent Tax Credit

### LINKED ISSUE #2401: Arizona Dependent Tax Credit
[Reference](https://azdor.gov/file/13752/download?token=zlZ1pofA) - page 21

### PR DESCRIPTION
**Pull request by PavelMakarchuk**
Fixes #2401

### Summary
🧒💰🔽

This pull request adds parameters for the Arizona dependent credit, a state income tax credit for taxpayers with dependents. The parameters include the credit amount and the reduction rate, which vary by the dependent's age, the taxpayer's income, and the tax year. The parameters are based on data from `policyengine_us/parameters/gov/states/az/tax/income/credits/dependent_credit/amount.yaml` and `policyengine_us/parameters/gov/states/az/tax/income/credits/dependent_credit/reduction/rate.yaml`.

> _`AZ` tax credit_
> _For dependents of some age_
> _Cut by income, fall_

### Walkthrough
*  Add parameters for the Arizona dependent credit, which is a tax credit for taxpayers who have dependents under a certain age ([link](https://github.com/PolicyEngine/policyengine-us/pull/2403/files?diff=unified&w=0#diff-e741ad0bbf94c77c67dae9175bfbf8d5e0d4dacd4c622b2f59438afc1e62a861R1-R22), [link](https://github.com/PolicyEngine/policyengine-us/pull/2403/files?diff=unified&w=0#diff-51d99be49a525dc4a22b129b66d077eae1bd14a6316f320421ae019b586f0308L1-R94))
  - Define the credit amount based on the dependent's age and the tax year in `amount.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2403/files?diff=unified&w=0#diff-e741ad0bbf94c77c67dae9175bfbf8d5e0d4dacd4c622b2f59438afc1e62a861R1-R22))
  - Define the reduction rate based on the taxpayer's adjusted gross income and the tax year in `reduction/rate.yaml` ([link](https://github.com/PolicyEngine/policyengine-us/pull/2403/files?diff=unified&w=0#diff-51d99be49a525dc4a22b129b66d077eae1bd14a6316f320421ae019b586f0308L1-R94))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/az/tax/income/credits/dependent_credit/reduction/credit_rate.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/az/tax/income/credits/dependent_credit/reduction/credit_rate.yaml:1]: @MaxGhenis we have modeled this as a single_amount parameter to represent the 5% every $1,000 from the law. Let me know if this works

File: policyengine_us/tests/policy/baseline/gov/states/az/tax/income/credits/dependent_credit/az_dependent_tax_credit.yaml [hunk 1]
```diff
@@ -0,0 +1,170 @@
+- name: Housheold with no dependents
```
Comment (MaxGhenis) [inline policyengine_us/tests/policy/baseline/gov/states/az/tax/income/credits/dependent_credit/az_dependent_tax_credit.yaml:1]:
```suggestion
- name: Household with no dependents
```

File: policyengine_us/variables/gov/states/az/tax/income/credits/dependent_credit/az_dependent_tax_credit.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        age = person("age", period)
+        dependent_amount = p.amount.calc(age) * dependent
+        amount = tax_unit.sum(dependent_amount)
+        income = tax_unit("adjusted_gross_income", period)
+        filing_status = tax_unit("filing_status", period)
+        reduction_start = p.reduction.start[filing_status]
+        excess = max_(income - reduction_start, 0)
+        increments = np.ceil(excess / p.reduction.increment)
+        reduction_percentage = increments * p.reduction.percentage
+        return max_(amount - amount * reduction_percentage, 0)
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/credits/dependent_credit/az_dependent_tax_credit.py:28]:
```suggestion
        return amount * (1 - reduction_percentage)
```

File: policyengine_us/variables/gov/states/az/tax/income/credits/dependent_credit/az_dependent_tax_credit.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        dependent = person("is_tax_unit_dependent", period)
+        age = person("age", period)
+        dependent_amount = p.amount.calc(age) * dependent
+        amount = tax_unit.sum(dependent_amount)
+        income = tax_unit("adjusted_gross_income", period)
+        filing_status = tax_unit("filing_status", period)
+        reduction_start = p.reduction.start[filing_status]
+        excess = max_(income - reduction_start, 0)
+        increments = np.ceil(excess / p.reduction.increment)
+        reduction_percentage = increments * p.reduction.percentage
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/az/tax/income/credits/dependent_credit/az_dependent_tax_credit.py:27]:
```suggestion
        reduction_percentage = min_(increments * p.reduction.percentage, 1)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**